### PR TITLE
:feet: Add analytics tracking code

### PR DESCRIPTION
**Pull request by neilbmclaughlin**
Analytics ID's need to be populated in env vars for the client code to
be added.

Changed config so it is the same for each environment and differences
are managed through env vars (in line with 12Factor principles)

Fixed eslint warning regarding function name missing

To be considered before merge:

- Content security policy (e.g. https://github.com/helmetjs/csp)
- Assets from CDN
- Webpack usage

### CODE REVIEW

File: app/middleware/locals.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = (config) =>
+  (req, res, next) => {
+    /* eslint-disable no-param-reassign */
+    res.locals.GOOGLE_ANALYTICS_TRACKING_ID = config.googleAnalyticsId;
+    res.locals.WEBTRENDS_TRACKING_ID = config.webtrendsId;
+    res.locals.HOTJAR_TRACKING_ID = config.hotjarId;
+    res.locals.FONT_CDN_HOST = config.fontCdnHost;
```
Comment (st3v3nhunt) [inline app/middleware/locals.js]: This isn't mentioned in the `README.md`

File: .bithoundrc [hunk 2]
```diff
@@ -42,17 +42,13 @@
     "**/extjs/src/**",
     "**/extjs/welcome/**",
     "bower_components/**",
-    "**/assets/**"
+    "**/assets/**",
```
Comment (st3v3nhunt) [inline .bithoundrc]: Is this needed? What goes in `**/assets/**`?

File: .bithoundrc [hunk 3]
```diff
@@ -42,17 +42,13 @@
...
+    "public/js/tracking/**"
```
Comment (st3v3nhunt) [inline .bithoundrc]: Should this be a bit more wide ranging? Given that the dir will have generated assets in? Or maybe wait until that time i.e. when the assets are being generated?

File: app/middleware/locals.js (hunk 1 above)
Comment (neilbmclaughlin) [inline app/middleware/locals.js]: Good, spot toyed with adding it but decided it was out of scope and left in in error.

File: .bithoundrc (hunk 2 above)
Comment (neilbmclaughlin) [inline .bithoundrc]: Nothing anymore so can be removed - will do this.

File: .bithoundrc (hunk 3 above)
Comment (neilbmclaughlin) [inline .bithoundrc]: Wait until that time.

Comment (neilbmclaughlin) [inline .bithoundrc]: webtrends.js replaced by webtrends.min.js

Comment (neilbmclaughlin) [inline .bithoundrc]: webtrends code moved into a public/js/vendor folder so we can be specific about not doing code analysis on 3rd party code.

File: .bithoundrc [hunk 4]
```diff
@@ -42,17 +42,12 @@
...
     "**/tests/**",
     "**/spec/**",
     "**/specs/**"
-  ],
-  "dependencies": {
-    "unused-ignores": [
-      "webpack"
-    ]
-  }
+  ]
```
Comment (neilbmclaughlin) [inline .bithoundrc]: This trailing comma was causing bithound excludes to fail.

**Comment by neilbmclaughlin** [comment]
Suggested changes made - this is good to go.

File: README.md [hunk 5]
```diff
@@ -39,6 +39,18 @@ is used to check all are present as part of the application start-up. If
...
 
+Environment variables are used to set application level settings for each
+environment.
+
+| Variable                        | Description                                                               | Default                  | Required |
+|:--------------------------------|:--------------------------------------------------------------------------|:-------------------------|----------|
+| `NODE_ENV`                      | node environment                                                          | development              | | 
+| `PORT`                          | server port                                                               | 3000                     | | 
+| `GOOGLE_ANALYTICS_TRACKING_ID`  | [Google Analytics](https://www.google.co.uk/analytics) property id        |                          | |
+| `WEBTRENDS_TRACKING_ID`         | [Webtrends](https://www.webtrends.com/) tracking id                       |                          | |
```
Comment (st3v3nhunt) [inline README.md]: In `config/config.js` this is `WEBTRENDS_ANALYTICS_TRACKING_ID`
and the hotJar one is `HOTJAR_ANALYTICS_TRACKING_ID`

Not sure if you want to change it here or in config?

Comment (neilbmclaughlin) [inline README.md]: Good spot! I'll change the readme.

Comment (st3v3nhunt) [inline README.md]: Ha yeah. I was going to mention this in the first place...and the possibility of using the require env vars package but as these aren't required that ins't going to work.

File: app/views/layout.nunjucks [hunk 6]
```diff
@@ -24,6 +24,21 @@
     <!--[if IE 8]><link href="./css/nhsuk-ie8.css" media="screen" rel="stylesheet" type="text/css" /><![endif]-->
 
     <link rel="stylesheet" href="./css/print.css" media="print" type="text/css">
+
+    {% if HOTJAR_TRACKING_ID %}
+       <!-- rational for this code being in head: https://docs.hotjar.com/v1.0/docs/manual -->
+      <script>
+        (function(h,o,t,j,a,r){
+        h.hj=h.hj||function(){(h.hj.q=h.hj.q||[]).push(arguments)};
+        h._hjSettings={hjid:{{ HOTJAR_TRACKING_ID }},hjsv:5};
```
Comment (st3v3nhunt) [inline app/views/layout.nunjucks]: The var needs escaping e.g.
`h._hjSettings={hjid:'{{ HOTJAR_TRACKING_ID }}',hjsv:5};`

Comment (st3v3nhunt) [inline app/views/layout.nunjucks]: No it doesn't - See this https://docs.hotjar.com/v1.0/docs/hotjar-tracking-code
(which I would have seen if I've read the comment!).
Hotjar IDs are numeric or at least need to be if they are to work but that is something provided by Hotjar...

**Comment by st3v3nhunt** [comment]
Longest PR EVER 😜

**Pull request merged**